src/lib/vip-import-validate-files.ts

```diff
@@ -1,122 +1,151 @@
-// Accepted media file extensions
-export const acceptedExtensions: string[] = [
-	'jpg',
-	'jpeg',
-	'jpe',
-	'gif',
-	'png',
-	'bmp',
-	'svg',
-	'tiff',
-	'tif',
-	'ico',
-	'asf',
-	'asx',
-	'wmv',
-	'wmx',
-	'wm',
-	'avi',
-	'divx',
-	'mov',
-	'qt',
-	'mpeg',
-	'mpg',
-	'mpe',
-	'mp4',
-	'm4v',
-	'ogv',
-	'webm',
-	'mkv',
-	'3gp',
-	'3gpp',
-	'3g2',
-	'3gp2',
-	'txt',
-	'asc',
-	'c',
-	'cc',
-	'h',
-	'srt',
-	'csv',
-	'tsv',
-	'ics',
-	'rtx',
-	'css',
-	'vtt',
-	'dfxp',
-	'mp3',
-	'm4a',
-	'm4b',
-	'ra',
-	'ram',
-	'wav',
-	'ogg',
-	'oga',
-	'mid',
-	'midi',
-	'wma',
-	'wax',
-	'mka',
-	'rtf',
-	'js',
-	'pdf',
-	'class',
-	'psd',
-	'xcf',
-	'doc',
-	'pot',
-	'pps',
-	'ppt',
-	'wri',
-	'xla',
-	'xls',
-	'xlt',
-	'xlw',
-	'mdb',
-	'mpp',
-	'docx',
-	'docm',
-	'dotx',
-	'dotm',
-	'xlsx',
-	'xlsm',
-	'xlsb',
-	'xltx',
-	'xltm',
-	'xlam',
-	'pptx',
-	'pptm',
-	'ppsx',
-	'ppsm',
-	'potx',
-	'potm',
-	'ppam',
-	'sldx',
-	'sldm',
-	'onetoc',
-	' onetoc2',
-	'onetmp',
-	'onepkg',
-	'oxps',
-	'xps',
-	'odt',
-	'odp',
-	'ods',
-	'odg',
-	'odc',
-	'odb',
-	'odf',
-	'webp',
-	'wp',
-	'wpd',
-	'key',
-	'numbers',
-	'pages',
-];
+/**
+ * Internal dependencies
+ */
+import { AppEnvironmentMediaImportConfig } from '../graphqlTypes';
+
+export const enum ValidateFilesErrors {
+	INVALID_TYPES = 'invalid_types',
+	INTERMEDIATE_IMAGES = 'intermediate_images',
+	INVALID_SIZES = 'invalid_sizes',
+	INVALID_NAMES = 'invalid_names',
+	INVALID_NAME_CHARACTER_COUNTS = 'invalid_name_character_counts',
+}
+
+interface LogErrorOptions {
+	errorType: ValidateFilesErrors;
+	invalidFiles: string[];
+	limit?: string | number | Record< string, string >;
+}
+
+interface ExtType {
+	ext: string | null;
+	type: string | null;
+}
+
+interface MediaImportAllowedFileTypes {
+	[ key: string ]: unknown;
+}
+
+interface ValidationResult {
+	intermediateImagesTotal: number;
+	errorFileTypes: string[];
+	errorFileNames: string[];
+	errorFileSizes: string[];
+	errorFileNamesCharCount: string[];
+	intermediateImages: { [ key: string ]: string };
+}
+
+/**
+ * File info validation
+ *
+ * Validate the file info for media files
+ */
+export async function validateFiles(
+	files: string[],
+	mediaImportConfig: AppEnvironmentMediaImportConfig
+): Promise< ValidationResult > {
+	const validationResult: ValidationResult = {
+		intermediateImagesTotal: 0,
+		errorFileTypes: [],
+		errorFileNames: [],
+		errorFileSizes: [],
+		errorFileNamesCharCount: [],
+		intermediateImages: {},
+	};
+
+	const fileValidationPromises = files.map( async file => {
+		const isFolder = await isDirectory( file );
+		const fileExtType = getFileExtType(
+			file,
+			<MediaImportAllowedFileTypes>mediaImportConfig.allowedFileTypes
+		);
+
+		if ( isInvalidFile( fileExtType, isFolder ) ) {
+			validationResult.errorFileTypes.push( file );
+		}
+
+		if ( ! isFileSizeValid( file, <number>mediaImportConfig.fileSizeLimitInBytes ) ) {
+			validationResult.errorFileSizes.push( file );
+		}
+
+		if ( isFileSanitized( file ) ) {
```

Comment:
Is this aligned with what we do in MMS?

Reply:
AFAIK, MMS doesn't validate it.

Reply:
I guess it wouldn't hurt to leave that in place.

Reply:
But we should align the two at some points.